docs/source/programming/vscode-with-slurm-job.md

```diff
@@ -0,0 +1,73 @@
+# Using VSCode with SLURM-managed jobs on the SWC HPC cluster
+
+This guide explains how to set up and use VSCode within a SLURM-managed job on the SWC HPC cluster, offering a solution for users who require fast access to shared storage or substantial computational resources.
+
+This solution is easy to set up and constrained by the SLURM job's resource limits, ensuring that users can work efficiently without overloading the system.
+
+
+## Instructions
+
+First, open a terminal (not the VSCode terminal) and connect to the gateway node by running:
+
+```{code-block} console
+$ ssh <SWC-USERNAME>@ssh.swc.ucl.ac.uk
+$ ssh hpc-gw1
+```
+
+Once connected, request an interactive job via SLURM to access a compute node. For example:
+
+```{code-block} console
+$ srun -p fast -n 4 --mem 8G --pty bash -i
+```
+
+In this example, `-p fast` requests the fast partition, with default time settings, though you may adjust this according to your needs. For more information, see the [SLURM arguments primer](https://howto.neuroinformatics.dev/programming/SLURM-arguments.html).
+
+After connecting to a compute node, initiate VSCode Code Tunnel by typing:
+
+```{code-block} console
+$ code tunnel
+```
+
+A URL will appear in the terminal, `https://github.com/login/device`.
+Follow this link, log in with your GitHub credentials, and enter the provided PIN to authorize access.
+
+You have two options to run VSCode:
+
+-  **Run VSCode in  the browser:**
+    After completing the above step, a second link will appear in the terminal (e.g., `https://vscode.dev/tunnel/gpu-XXXX`), which you can follow to access your VSCode session running directly on the HPC compute node.
+
+- **Run VSCode on your local machine:**
+    If you want instead to use your local VSCode, install the "Remote - Tunnels" extension, click on "Open remote window" in the bottom left corner of the VSCode window, and select "Connect to Tunnel". You should see the name "gpu-XXXX" in the list of available tunnels. Click on it to connect to the VSCode session running on the HPC compute node.
+
+NB: the GPU name in the URL provided might not match the actual node name.
```

Comment:
```suggestion
NB: the node name in the provided URL might not match the actual node name.
```

Let's stick with the "node name" terminology without specific reference to `gpu` (see comment above).